server/src/test/java/org/elasticsearch/cluster/metadata/IndexCreationTaskTests.java

```diff
@@ -491,5 +491,7 @@ private void setupIndicesService() throws Exception {
         when(service.getIndexEventListener()).thenReturn(mock(IndexEventListener.class));
 
         when(indicesService.createIndex(anyObject(), anyObject())).thenReturn(service);
+        when(allocationService.reroute(any(ClusterState.class), anyString())).thenAnswer(
```

Comment:
I am unclear on why this needed to be added. I mean it seems necessary now, but why was it not necessary before?

Reply:
Oh nvm you already answered this in your original post.